darshan-runtime/lib/darshan-pnetcdf-api.m4

```diff
@@ -234,7 +234,7 @@ int DARSHAN_DECL(APINAME($1,$2,$3,$4))(int ncid, int varid, ArgKind($2)BufArgs($
     if (ret == NC_NOERR) {
         PNETCDF_VAR_PRE_RECORD();
         struct pnetcdf_var_record_ref *rec_ref;
-        rec_ref = darshan_lookup_record_ref(pnetcdf_var_runtime->ncid_hash, &ncid, sizeof(int));
+        rec_ref = darshan_lookup_record_ref(pnetcdf_var_runtime->ncid_hash, &varid, sizeof(int));
```

Comment:
This looks odd to look up `varid` in something called `ncid_hash`.  Is this because the ncid_hash data structure is doing double duty to track both depending on whether it is instrumentation for the pnetcdf file module or the pnetcdf var module?

Reply:
That's a good point. I was thinking that maybe `ncid` was a more general thing like a `hid` is in HDF5 (a `hid` could reference an open file, dataset, property list, etc.), but looking closer at the PnetCDF interface docs it's not. An `ncid` (which is just an int) refers specifically to an open file it looks like, so I see how that could get confusing. I'll change the name here to fix that up, too.

Reply:
They are not in the same hash. Looking closer at the code, I think it was just copying the HDF5 module where the same exact runtime structure could be used for files and datasets since everything is an `hid`. That works fine with the fix applied previously here, but it is confusing so I did try to fix. Maybe sanity check that  8538154 looks okay?